### PR TITLE
feat: Add migration for default Guardian Ideal widget

### PR DESCRIPTION
**Pull request by abbyoung**
# SC-1587

## Proposed changes

* Adds migration that updates MySpace to include Guardian Ideal widget by default.
* Adds tests for migration
* Adds default widgets when new users are created
* Updates tests for default widgets

## Reviewer notes

If you want to test _only_ this migration via CLI, pass in the migration name when running migrate down. This is ideal, since all other migrations have already run on prod. 

`yarn migrate down 1680127440523-add-guardian-ideal.js`

* If a user has the FeaturedShortcuts widget in their MySpace, GuardianIdeal should be added as the second item.
* If a user does not have FeaturedShortcuts, GuardianIdeal should appear as the first item.
* If a user has already added GuardianIdeal to their MySpace, nothing should happen. The widget in MySpace should be missing `default: true` in the database.

## Setup

Try running the app and verify GuardianIdeal widget appears in MySpace.

To experiment with edge cases, pull up Mongo Express on `localhost:8888` and use the yarn migrate up/down commands to trigger the migrations.

### As the original developer, I have

- [ ] Met the acceptance criteria, or will meet them in subsequent PRs or stories
  - ... 
- [ ] Created new stories in Storybook if applicable
  - [ ] Checked that all Storybook accessibility checks are passing
- [ ] Created/modified automated unit tests in Jest
  - [ ] Including jest-axe checks when UI changes
- [ ] Created/modified automated E2E tests in Cypress
  - [ ] Including cypress-axe checks when UI changes
  - [ ] Checked that the E2E test build is not failing
- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):
  - [ ] Checked responsiveness in mobile, tablet, and desktop
  - [ ] Checked keyboard navigability
  - [ ] Tested with [VoiceOver](https://dequeuniversity.com/screenreaders/voiceover-keyboard-shortcuts) in Safari
  - [ ] Checked VO's [rotor menu](https://github.com/trussworks/accessibility/blob/master/README.md#how-to-use-the-rotor-menu) for landmarks, page heading structure and links
  - [ ] Used a browser a11y tool to check for issues (WAVE, axe, ANDI or Accessibility addon tab for Storybook)
- [ ] Requested a design review for user-facing changes
- For any new migrations/schema changes:
  - [ ] Followed guidelines for zero-downtime deploys

### As code reviewer(s), I have

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
  - [ ] Checked that the E2E test build is not failing
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] Considered marking this as accepted even if there are small changes needed
- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):
  - [ ] Checked responsiveness in mobile, tablet, and desktop
  - [ ] Checked keyboard navigability
  - [ ] Tested with [VoiceOver](https://dequeuniversity.com/screenreaders/voiceover-keyboard-shortcuts) in Safari
  - [ ] Checked VO's [rotor menu](https://github.com/trussworks/accessibility/blob/master/README.md#how-to-use-the-rotor-menu) for landmarks, page heading structure and links
  - [ ] Used a browser a11y tool to check for issues (WAVE, axe, ANDI or Accessibility addon tab for Storybook)

### As a designer reviewer, I have

- [ ] Checked in the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow
- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):
  - [ ] Checked responsiveness in mobile, tablet, and desktop
  - [ ] Checked keyboard navigability
  - [ ] Tested with [VoiceOver](https://dequeuniversity.com/screenreaders/voiceover-keyboard-shortcuts) in Safari
  - [ ] Checked VO's [rotor menu](https://github.com/trussworks/accessibility/blob/master/README.md#how-to-use-the-rotor-menu) for landmarks, page heading structure and links
  - [ ] Used a browser a11y tool to check for issues (WAVE, axe, ANDI or Accessibility addon tab for Storybook)

### As a test user, I have

- Run through the [Test Script](hhttps://docs.google.com/spreadsheets/d/1eV8UK0aJZ0qrzjnXf5SJ_uRiV7bpsj3yrhRFEkjOvV4/edit?usp=sharing):
  - [ ] On commercial internet in IE11
  - [ ] On commercial internet in Firefox
  - [ ] On commercial internet in Chrome
  - [ ] On commercial internet in Safari
  - [ ] On NIPR in IE11
  - [ ] On NIPR in Firefox
  - [ ] On NIPR in Chrome
  - [ ] On NIPR in Safari
  - [ ] On a mobile device in Firefox
  - [ ] On a mobile device in Chrome
  - [ ] On a mobile device in Safari
- [ ] Added any anomalous behavior to this PR

---

### CODE REVIEW

File: src/components/MySpace/MySpace.tsx [hunk 1]
```diff
@@ -83,7 +83,7 @@ const MySpace = ({ bookmarks }: { bookmarks: BookmarkRecords }) => {
     )
 
     handleAddWidget({
-      variables: { title: 'Ideal Title', type: AddWidgetType.GuardianIdeal },
+      variables: { title: 'Guardian Ideal', type: AddWidgetType.GuardianIdeal },
```
Comment (abbyoung) [inline src/components/MySpace/MySpace.tsx:86]: Updated this to match pattern for the other widgets. An objection to the title of this widget being "Guardian Ideal"?

File: migrations/1680127440523-add-guardian-ideal.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    }
+  )
+
+  // If the first item is not FeaturedShortcuts or GuardianIdeal,
+  // add GuardianIdeal as the first item in a user's MySpace.
+  await db.collection('users').updateMany(
+    {
+      'mySpace.0.type': {
+        //$nin checks that the value is not in the array of values provided
+        $nin: ['GuardianIdeal', 'FeaturedShortcuts'],
```
Comment (gidjin) [inline migrations/1680127440523-add-guardian-ideal.js:44]: Question: Is it worth it to add both in this migration if the user doesn't have either? Now that we add it for new users we'll need to have a migration to add `FeaturedShortcuts` again, but I'm fine if we want to keep that separate from this one.

**Review by gidjin — APPROVED** [review]
LGTM, left a question about Featured Shortcuts but I'm fine if that's a separate PR.

**Review by jcbcapps — APPROVED** [review]
Looks great!

File: migrations/1680127440523-add-guardian-ideal.js (hunk 2 above)
Comment (abbyoung) [reply migrations/1680127440523-add-guardian-ideal.js:44]: let's keep that as a separate PR, just to keep the migrations a bit simpler. i feel like this PR has gotten a bit more complex than the original story as is.

**Pull request merged**